scripts/deployment.ts

```diff
@@ -0,0 +1,156 @@
+  const validatorWalletCreator = await deployContract("ValidatorWalletCreator", signer);
+  const rollupCreator = await deployContract("RollupCreator", signer);
+  return {
+    bridgeCreator,
+    prover0,
+    proverMem,
+    proverMath,
+    proverHostIo,
+    osp,
+    challengeManager,
+    rollupAdmin,
+    rollupUser,
+    validatorUtils,
+    validatorWalletCreator,
+    rollupCreator,
+  };
+}
+
+async function main() {
+
+  // Get the private key and RPC to deploy the contract
+  const privateKey = process.env.PRIVATE_KEY;
+  if (!privateKey) {
+    throw new Error("Private key is not defined");
+  }
+
+  const providerAPI = process.env.RPC_URL;
+  if (!providerAPI) {
+    throw new Error("RPC URL is not defined");
+  }
+
+  const provider = new providers.JsonRpcProvider(providerAPI)
+  const signer = new Wallet(privateKey, provider)
+
+  try {
+
+    // Deploying all contracts
+    const contracts = await deployAllContracts(signer);
+    
+    /* 
+      * Call setTemplates with the deployed contract addresses
+      * Adding 15 million gas limit otherwise it'll be reverted with "gas exceeds block gas limit" error
+    */
+    console.log("Waiting for the Template to be set on the Rollup Creator")
+    await contracts.rollupCreator.setTemplates(
+      contracts.bridgeCreator.address,
+      contracts.osp.address,
+      contracts.challengeManager.address,
+      contracts.rollupAdmin.address,
+      contracts.rollupUser.address,
+      contracts.validatorUtils.address,
+      contracts.validatorWalletCreator.address
+      , {gasLimit: ethers.BigNumber.from("15000000")}
+    );
+    console.log("Template is set on the Rollup Creator")
+
+    // Define the configuration for the createRollup function
+    const rollupConfig = {
+      confirmPeriodBlocks: 20,
+      extraChallengeTimeBlocks: 0,
+      stakeToken: ethers.constants.AddressZero,
+      baseStake: ethers.utils.parseEther("1"),
+      wasmModuleRoot: "0xda4e3ad5e7feacb817c21c8d0220da7650fe9051ece68a3f0b1c5d38bbb27b21",
+      owner: signer.address,
+      loserStakeEscrow: ethers.constants.AddressZero,
+      chainId: 421613,
+      chainConfig:ethers.constants.HashZero,
+      genesisBlockNum: 0,
+      sequencerInboxMaxTimeVariation: {
+        delayBlocks: 16,
+        futureBlocks: 192,
+        delaySeconds: 86400,
+        futureSeconds: 7200,
+      },
+    };
+    
+
+    /* 
+      * Call the createRollup function
+      * Adding 15 million gas limit otherwise it'll be reverted with "gas exceeds block gas limit" error
+    */
+   console.log("Calling createRollup to generate a new rollup ...")
+    const createRollupTx = await contracts.rollupCreator.createRollup(rollupConfig, {gasLimit: ethers.BigNumber.from("15000000")});
+    const createRollupReceipt = await createRollupTx.wait();
+    const rollupCreatedEvent = createRollupReceipt.events?.find(
+      (event: { event: string }) => event.event === "RollupCreated"
+      );
```

Comment:
should also match the event address == rollupCreator

Reply:
I think using this safer, in case we add other events?

Reply:
Not sure what u mean, at current state it might be possible to fake a RollupCreated event from another contract.

Reply:
added check here Harry:
https://github.com/OffchainLabs/nitro-contracts/pull/16